airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSourceOperations.java

```diff
@@ -80,8 +95,12 @@ public void setJsonField(final ResultSet resultSet, final int colIndex, final Ob
       putBoolean(json, columnName, resultSet, colIndex);
     } else if (columnTypeName.equalsIgnoreCase("bytea")) {
       putString(json, columnName, resultSet, colIndex);
-    } else if (columnTypeName.equalsIgnoreCase("time") || columnTypeName.equalsIgnoreCase("timetz")) {
-      putString(json, columnName, resultSet, colIndex);
+    } else if (columnTypeName.equalsIgnoreCase("time")) {
+      putTime(json, columnName, resultSet, colIndex);
+    } else if (columnTypeName.equalsIgnoreCase("timetz")) {
+      putTimeWithTimezone(json, columnName, resultSet, colIndex);
+    } else if (columnTypeName.equalsIgnoreCase("timestamptz")) {
+      putTimestampWithTimezone(json, columnName, resultSet, colIndex);
```

Comment:
Is it possible we will have more `else if` statements here, in the future?
I suggest refactoring this block to smth like:
```java
 return switch (typeName) {
      case "real", "double", "float" -> Double.parseDouble(varCharValue);
      case "varchar" -> varCharValue;
      case "boolean" -> Boolean.parseBoolean(varCharValue);
      case "integer" -> Integer.parseInt(varCharValue);
      default -> null;
    };
```

__
[![sema-logo](https://app.semasoftware.com/img/sema-tray-logo.gif)](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix

Reply:
> Is it possible we will have more `else if` statements here, in the future? I suggest refactoring this block to smth like:
> 
> ```java
>  return switch (typeName) {
>       case "real", "double", "float" -> Double.parseDouble(varCharValue);
>       case "varchar" -> varCharValue;
>       case "boolean" -> Boolean.parseBoolean(varCharValue);
>       case "integer" -> Integer.parseInt(varCharValue);
>       default -> null;
>     };
> ```
> 
> __ [![sema-logo](https://camo.githubusercontent.com/843687ee2a84b8cc104a1e4297c26c9db3d3f710080228887db6338b4550a240/68747470733a2f2f6170702e73656d61736f6674776172652e636f6d2f696d672f73656d612d747261792d6c6f676f2e676966)](https://semasoftware.com/gh) [ ![sema-logo](https://camo.githubusercontent.com/843687ee2a84b8cc104a1e4297c26c9db3d3f710080228887db6338b4550a240/68747470733a2f2f6170702e73656d61736f6674776172652e636f6d2f696d672f73656d612d747261792d6c6f676f2e676966) ](https://semasoftware.com/gh) [ ](https://semasoftware.com/gh)  **Summary:** 🛠️ This code needs a fix

replaced with switch